libraries/AP_Airspeed/AP_Airspeed_UAVCAN.cpp

```diff
@@ -37,6 +37,7 @@ void AP_Airspeed_UAVCAN::subscribe_msgs(AP_UAVCAN* ap_uavcan)
 
     const int airspeed_listener_res = airspeed_listener->start(AirspeedCb(ap_uavcan, &handle_airspeed));
     if (airspeed_listener_res < 0) {
+        delete airspeed_listener;
```

Comment:
it makes no sense to free memory before we panic. It just makes for fatter firmware with no benefit